src/chatgpt.py

```diff
@@ -316,9 +391,22 @@ def start_prompt(
         body["response_format"] = {"type": "json_object"}
 
     try:
-        r = requests.post(
-            f"{BASE_ENDPOINT}/chat/completions", headers=headers, json=body
-        )
+        if config["supplier"] == "azure":
+            headers = {
+                "Content-Type": "application/json",
+                "api-key": api_key,
+            }
+            r = requests.post(
+                f"{base_endpoint}/openai/deployments/{model}/chat/completions?api-version={api_version}", headers=headers, json=body
+            )
+        elif config["supplier"] == "openai":
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {api_key}",
+            }
+            r = requests.post(
+                f"{base_endpoint}/chat/completions", headers=headers, json=body
+            )
```

Comment:
More switching on the `supplier` - makes me think that having classes as mentioned before will be nice.